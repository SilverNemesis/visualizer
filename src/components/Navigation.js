import React from 'react'
import { NavBar, NavBarToggler, NavBarNav } from '../primitives'
import { Link } from 'react-router-dom'

class Navigation extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            expanded: false
        }
        this.onClickExpand = this.onClickExpand.bind(this)
    }

    onClickExpand() {
        this.setState({
            expanded: !this.state.expanded
        })
    }

    render() {
        return (
            <NavBar large dark>
                <Link to="/" className="navbar-brand"><h1>Visualizer</h1></Link>
                <NavBarToggler expanded={this.state.expanded} onClick={this.onClickExpand}>
                    <NavBarNav>
                        <Link to="/" className="nav-link">Home</Link>
                        <Link to="/sort" className="nav-link">Sort</Link>
                        <Link to="/maze" className="nav-link">Maze</Link>
                    </NavBarNav>
                </NavBarToggler>
            </NavBar>
        )
    }
}

export default Navigation