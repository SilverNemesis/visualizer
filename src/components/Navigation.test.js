import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Navigation from './Navigation'

const renderer = new ShallowRenderer();

it('renders correctly', () => {
    renderer.render(<Navigation />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
});