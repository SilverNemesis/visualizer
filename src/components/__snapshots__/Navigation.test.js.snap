// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<NavBar
  dark={true}
  large={true}
>
  <ForwardRef
    className="navbar-brand"
    to="/"
  >
    <h1>
      Visualizer
    </h1>
  </ForwardRef>
  <NavBarToggler
    expanded={false}
    onClick={[Function]}
  >
    <NavBarNav>
      <ForwardRef
        className="nav-link"
        to="/"
      >
        Home
      </ForwardRef>
      <ForwardRef
        className="nav-link"
        to="/sort"
      >
        Sort
      </ForwardRef>
      <ForwardRef
        className="nav-link"
        to="/maze"
      >
        Maze
      </ForwardRef>
      <ForwardRef
        className="nav-link"
        to="/tree"
      >
        Tree
      </ForwardRef>
    </NavBarNav>
  </NavBarToggler>
</NavBar>
`;
